src/commands/community/nft/query.ts

```diff
@@ -79,6 +101,294 @@ const txHistoryEmojiMap: Record<string, string> = {
   listing: getEmoji("listing"),
 }
 
+function buildSwitchViewActionRow(
+  currentView: string,
+  symbol: string,
+  collectionAddress: string,
+  tokenId: string,
+  chain: string
+) {
+  const nftButton = new MessageButton({
+    label: "NFT",
+    emoji: "🖼",
+    customId: `nft-view/nft/${symbol}/${collectionAddress}/${tokenId}/${chain}`,
+    style: "SECONDARY",
+    disabled: currentView === "nft",
+  })
+  const tickerButton = new MessageButton({
+    label: "Ticker",
+    emoji: "📈",
+    customId: `nft-view/ticker/${symbol}/${collectionAddress}/${tokenId}/${chain}`,
+    style: "SECONDARY",
+    disabled: currentView === "ticker",
+  })
+  const collectionInfoButton = new MessageButton({
+    label: "Collection Info",
+    emoji: "🔎",
+    customId: `nft-view/info/${symbol}/${collectionAddress}/${tokenId}/${chain}`,
+    style: "SECONDARY",
+    disabled: currentView === "info",
+  })
+  const row = new MessageActionRow()
+  row.addComponents([nftButton, tickerButton, collectionInfoButton])
+  return row
+}
+
+function collectButton(msg: Message, originMsg: Message) {
+  return msg
+    .createMessageComponentCollector({
+      componentType: MessageComponentTypes.BUTTON,
+      idle: 60000,
+      filter: filter(originMsg.author.id),
+    })
+    .on("collect", async (i) => {
+      await switchView(i, originMsg)
+    })
+    .on("end", () => {
+      msg.edit({ components: [] }).catch(() => null)
+    })
+}
+
+async function switchView(i: ButtonInteraction, msg: Message) {
```

Comment:
add i.deferUpdate() since we edit original message

Reply:
no need since I already call defer in the filter function